src/v/crypto/random.cc

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "crypto/crypto.h"
+#include "crypto/ssl_utils.h"
+#include "internal.h"
+
+#include <openssl/rand.h>
+
+namespace crypto {
+bytes_span<> generate_random(bytes_span<> buf, use_private_rng private_rng) {
+    if (private_rng) {
+        if (1 != RAND_priv_bytes_ex(nullptr, buf.data(), buf.size(), 0)) {
+            throw internal::ossl_error(
+              "Failed to get random output from private RNG");
+        }
+    } else {
+        if (1 != RAND_bytes_ex(nullptr, buf.data(), buf.size(), 0)) {
```

Comment:
What does it mean for "The bytes generated will have a security strength of at least strength bits", when strength is 0?

Reply:
I believe '0' means "don't care" (it's not documented at all... `RAND_bytes` calls `RAND_bytes_ex` with a strength of 0).  The strength indicates an expected level of security.  So if the DRBG supports 128 bits of security but the caller requests 256, it will fail.  I don't believe it's necessarily a parameter to condition the DRBG output but just to ensure it meets a minimum level.

But I will look into it.

Reply:
Yeah confirmed.... strength is just a check to ensure the RNG being used can supply the appropriate security level, per their docs:

> The bytes generated will have a security strength of at least strength bits.

Reply:
It's still not clear to me what it is. This is about as far as I got trying to figure it out: https://github.com/openssl/openssl/pull/15560